components/forms/FormSubmitFormFinal.tsx

```diff
@@ -0,0 +1,181 @@
+
+import { submitForm } from '~/actions/form.actions';
+import { validateResolver } from '~/lib/validateResolver';
+
+import {
+  type ElementsType,
+  FormElements,
+  type validationProperty,
+} from './interfaces/FormElements';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from '../ui/form';
+import { Button } from '../ui/button';
+import { Separator } from '../ui/separator';
+import { toast } from '../ui/use-toast';
+
+interface FormSubmitFormProps {
+  form: {
+    id: number;
+    title: string;
+    description: string | undefined;
+    on_submit_message: string;
+    pages: number;
+  };
+  questions: {
+    id: string;
+    question: string;
+    description?: string | undefined;
+    required: boolean;
+    input_type: ElementsType;
+    items?: string[] | undefined;
+    mime_types?: string[] | undefined;
+    range?: string[] | undefined;
+    marks: number;
+  }[][];
+  requiredQuestions: string[];
+  questionValidations: Record<string, validationProperty>;
+  answers?: Record<string, string | number | string[]>;
+}
+export default function FormSubmitFormFinal({
+  form,
+  questions,
+  requiredQuestions,
+  questionValidations,
+  answers,
+}: FormSubmitFormProps) {
+  const Router = useRouter();
+  console.log(form.pages);
+  const [previousStep, setPreviousStep] = useState(0);
+  const [currentStep, setCurrentStep] = useState(0);
+  const delta = currentStep - previousStep;
+  const jsonSchema: {
+    type: string;
+    properties: Record<string, validationProperty>;
+    required: string[];
+    additionalProperties: boolean;
+  } = {
+    type: 'object',
+    properties: questionValidations,
+    required: requiredQuestions,
+    additionalProperties: false,
+  };
+  const resolver = (data: FormData, context: string | undefined) =>
+    validateResolver(data, context, jsonSchema);
+  const forms = useForm({
+    context: form.id.toString(),
+    resolver: resolver,
+    defaultValues: answers,
+  });
+  const next = async () => {
+    const fields = questions[currentStep].map((question) => question.id);
+    console.log(forms.getValues());
+    // types cannot be inferred during compile time
+    const output = await forms.trigger(fields as never, {
+      shouldFocus: true,
+    });
+    console.log(fields);
+    console.log(output);
+    if (!output) return;
+
+    if (currentStep < form.pages - 1) {
+      setPreviousStep(currentStep);
+      setCurrentStep((step) => step + 1);
+    }
+  };
+
+  const prev = () => {
+    if (currentStep > 0) {
+      setPreviousStep(currentStep);
+      setCurrentStep((step) => step - 1);
+    }
+  };
+  const onSubmit = async () => {
+    console.log(forms.getValues());
+    const output = await forms.trigger();
+    console.log(forms.getValues());
+    console.log(output);
+    if (!output) return;
+    const result = await submitForm(form.id, forms.getValues());
+    toast(result);
+    Router.push('/en/forms');
```

Comment:
Typo, I think you missed making this dynamic.
```suggestion
    Router.push(`/${locale}/forms`);
```